core/src/main/java/org/elasticsearch/index/query/functionscore/ScoreFunctionBuilders.java

```diff
@@ -93,4 +93,16 @@ public static WeightBuilder weightFactorFunction(float weight) {
     public static FieldValueFactorFunctionBuilder fieldValueFactorFunction(String fieldName) {
         return new FieldValueFactorFunctionBuilder(fieldName);
     }
+
+    public static void registerBuilders(SearchModule searchModule, NamedWriteableRegistry namedWriteableRegistry) {
```

Comment:
Yeah. I'm not sure where to put that method. I'm happy to move it though.
Or make a new object for it.

Having it in the same package let's us have more package private methods
though.
On Mar 31, 2016 7:51 AM, "Luca Cavanna" <notifications@github.com> wrote:

> In
> core/src/main/java/org/elasticsearch/index/query/functionscore/ScoreFunctionBuilders.java
> <https://github.com/elastic/elasticsearch/pull/17351#discussion_r58040617>
> :
>
> > @@ -93,4 +93,16 @@ public static WeightBuilder weightFactorFunction(float weight) {
> >      public static FieldValueFactorFunctionBuilder fieldValueFactorFunction(String fieldName) {
> >          return new FieldValueFactorFunctionBuilder(fieldName);
> >      }
> > +
> > +    public static void registerBuilders(SearchModule searchModule, NamedWriteableRegistry namedWriteableRegistry) {
>
> This feels a little out of place, as this class is here for the java api.
> I guess there was no better place to put it?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/elastic/elasticsearch/pull/17351/files/13b59d5c98f8bcd9ba66c5644d4500a95427b1a3#r58040617>
>